cpp/tests/t/geometry/TriangleMesh.cpp

```diff
@@ -941,5 +941,92 @@ TEST_P(TriangleMeshPermuteDevices, CreateMobius) {
             triangle_indices_custom));
 }
 
+TEST_P(TriangleMeshPermuteDevices, SelectByIndex_Box) {
```

Comment:
we can also add a test for an empty mesh, empty index list, index lists with negatives and duplicates.